packages/backend-tools/src/logger/ElasticSearchBackend.ts

```diff
@@ -0,0 +1,100 @@
+import { Client } from '@elastic/elasticsearch'
+import { v4 as uuidv4 } from 'uuid'
+
+import { LoggerBackend } from './interfaces'
+
+export interface ElasticSearchBackendOptions {
+  node: string
+  apiKey: string
+  flushInterval?: number
+  indexPrefix?: string
+}
+
+export class ElasticSearchBackend implements LoggerBackend {
+  private readonly options: Required<ElasticSearchBackendOptions>
+  private readonly buffer: string[]
+  private readonly client: Client
+
+  constructor(options: ElasticSearchBackendOptions) {
+    this.options = {
+      ...options,
+      flushInterval: options.flushInterval ?? 10000,
+      indexPrefix: options.indexPrefix ?? 'logs',
+    }
+
+    this.client = new Client({
+      node: options.node,
+      auth: {
+        apiKey: options.apiKey,
+      },
+    })
+
+    this.buffer = []
+    this.start()
+  }
+
+  public debug(message: string): void {
+    this.buffer.push(message)
+  }
+
+  public log(message: string): void {
+    this.buffer.push(message)
+  }
+
+  public warn(message: string): void {
+    this.buffer.push(message)
+  }
+
+  public error(message: string): void {
+    this.buffer.push(message)
+  }
+
+  private start(): void {
+    // eslint-disable-next-line @typescript-eslint/no-misused-promises
+    setInterval(async () => {
+      await this.flushLogs()
+    }, this.options.flushInterval)
+  }
+
+  private async flushLogs(): Promise<void> {
+    try {
+      const index = await this.createIndex()
+
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+      const documents = this.buffer.map((message) => ({
+        id: uuidv4(),
```

Comment:
Can we use [nanoid](https://www.npmjs.com/package/nanoid) instead? uuid is not cryptographically secure and is a pain to copy by double clicking, it takes more space and degrades database performance.

Reply:
Hmm wasn't aware of `nanoid` but on the other hand:
- double clicking is still an issue, even the example on npm site includes a dash ;)
- it doesn't have to be secure, we only need to be unique